packages/client/src/lib.rs

```diff
@@ -13,49 +15,80 @@ pub enum Error {
     QueryFailed(#[from] StdError),
 }
 
-pub struct Client<'a, M, Q>
+pub struct Client<'a, T = Empty> {
+    querier: QuerierWrapper<'a>,
+    custom_msg_type: PhantomData<T>,
+}
+
+impl<'a, T> Client<'a, T> {
+    pub fn new(querier: QuerierWrapper<'a>) -> Self {
+        Self {
+            querier,
+            custom_msg_type: PhantomData,
+        }
+    }
+
+    pub fn execute<M>(&self, msg: M) -> CosmosMsg<T>
+    where
+        M: Into<CosmosMsg<T>>,
+    {
+        msg.into()
+    }
+
+    pub fn query<R, Q, C>(&self, msg: Q) -> Result<R, Error>
+    where
+        R: DeserializeOwned,
+        Q: Into<QueryRequest<C>>,
+        C: CustomQuery,
+    {
+        QuerierWrapper::new(self.querier.deref())
+            .query(&msg.into())
+            .map_err(Into::into)
+            .map_err(Report::new)
+    }
+}
```

Comment:
this shouldn't be publicly accessible. The only way to access core should be through the `nexus::Client`